src/test/scala/zio/rocksdb/TransactionDBSpec.scala

```diff
@@ -72,8 +72,17 @@ object TransactionDBSpec extends DefaultRunnableSpec {
               }
           actual <- TransactionDB.get(key)
         } yield assert(actual.map(bytesToInt))(expected)
-      }
-    ) @@ timeout(1 second)
+      },
+      testM("thread safety inside transaction") {
+        checkM(byteArray, byteArray) { (b1, b2) =>
+          for {
+            _ <- TransactionDB.atomically {
+                  Transaction.put(b1, b2) <&> Transaction.put(b2, b1)
+                }
+          } yield assertCompletes
+        }
+      } @@ nonFlaky(10)
```

Comment:
this test would fail sometimes because of multiple threads trying to update the Transaction simultaneously.